Automattic/jetpack
# Instant Search: fix many theme incompatibilities

Note: this merges into the instant-search-master branch

Fixes many of the problems in #13391 particularly:
- inserting results into the theme works in all those themes
- the search form for all those themes works as search as you type as does the default core WP search widget
- Adds some CSS for our search box and cleans it up
- Adjust opacity when results are loading to give the user some feedback
- Starts the search query after only 200ms rather than 500ms. Feels a lot snappier
- Did some slight code cleanup and renamed the "widget" to "app" so it isn't confused with the WP widgets.

#### Testing instructions:
* Add define( "JETPACK_SEARCH_PROTOTYPE", true ); to your wp-config.php.
* Ensure that your site has the Jetpack Pro plan and has Jetpack Search enabled.
* Add a Jetpack Search widget to the Search page sidebar. (though I guess it should now work without this added at all)
* Enter a query into a search widget. Alternatively, navigate to a search page like /?s=privacy.
* You can also test against any other site by adding this filter and then adding `&blog_id=20115252` to your search page url (that is the jetpack.com blog_id).

```
function jp_instant_search_options( $options ) {
	if ( $_GET['blog_id'] ) {
		$options['siteId'] = (int) $_GET['blog_id'];
	}
	return $options;
}
add_filter( 'jetpack_instant_search_options', 'jp_instant_search_options' );
```

#### Proposed changelog entry for your changes:
* None
